.github/workflows/ci-cd.yml

```diff
@@ -73,20 +73,31 @@ jobs:
     runs-on: ubuntu-latest
     environment: test_release
     concurrency: release
+    if: github.ref_name != 'main'
 
     steps:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
           ref: ${{ github.head_ref || github.ref_name }}
 
-      # Do a dry run of PSR
+      # Dry run of PSR to build the distribution
       - name: Test release
         uses: python-semantic-release/python-semantic-release@v8.5.0
-        if: github.ref_name != 'main'
         with:
           root_options: --noop
 
+      - uses: snok/install-poetry@v1.3.4
+      - name: Install Dependencies
+        run: poetry install --only main,test_build
+        shell: bash
+
+      - name: Test build of distribution packages
+        shell: bash
+        run: |
+          poetry build
+          poetry run python -Im twine check --strict dist/*
+
   check_if_should_release:
```

Comment:
Just get rid of this job and move the `if`-checks to `release`.

Reply:
8a809d709d0db338f0213a80bbaf05e1ff7a795d